internal/datasets/subjectsetbyresourceid.go

```diff
@@ -36,12 +42,22 @@ func (ssr SubjectSetByResourceID) AddFromRelationship(relationship *core.Relatio
 }
 
 // UnionWith unions the map's sets with the other map of sets provided.
-func (ssr SubjectSetByResourceID) UnionWith(other map[string]*v1.FoundSubjects) {
+func (ssr SubjectSetByResourceID) UnionWith(other map[string]*v1.FoundSubjects) error {
```

Comment:
Can you create a test which reproduces why this happens in real running-systems to verify the fix?